app/collections/capability.js

```diff
@@ -1,71 +1,62 @@
-import {
-    unableToRetrieve,
-    saveFailed,
-    saveSuccessful,
-    updateFailed,
-    updateSuccessful
-} from '../notifications/general';
+import _ from 'lodash';
 
-const Datastore = require('nedb');
-const electron = require('electron');
-const path = require('path');
+import connectionToDB from './connectionSetup';
+import {displayToast} from '../notifications';
 
-const userDataPath = (electron.app || electron.remote.app).getPath('userData');
-const collectionsPath = path.join(userDataPath, 'collections');
-const Capability = new Datastore({
-    filename:              path.join(collectionsPath, 'question.db'),
-    autoload:              true,
-    corruptAlertThreshold: 1,
-    timestampData:         true
-});
+const Capability = connectionToDB('question');
 
-export const getAllQuestions = () => new Promise(resolve => Capability.find({}, (err, docs) => {
-    if (err) {
-        unableToRetrieve();
-    }
+export async function getAllQuestions() {
+    try {
+        const result = await Capability.find({});
 
-    return resolve(docs);
-}));
+        return result;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
 
-const addNewQuestion = data => {
-    Capability.insert(data, error => {
-        if (error) {
-            saveFailed();
-        }
-        saveSuccessful();
-    });
-};
+        return null;
+    }
+}
+
+async function addNewQuestion(data) {
+    try {
+        await Capability.insert(data);
+        displayToast('saveSuccess');
+    } catch (e) {
+        displayToast('saveFail');
+        console.log(e);
+    }
+}
 
-const updateQuestion = data => {
+async function updateQuestion(data) {
     const {classroomId, questionSet} = data;
+    try {
+        await Capability.update({classroomId}, {$set: {questionSet}}, {});
 
-    Capability.update({classroomId}, {$set: {questionSet}}, {}, err => {
-        if (err) {
-            updateFailed();
-        }
-        updateSuccessful();
-    });
-};
+        displayToast('updateSuccess');
+    } catch (e) {
+        displayToast('updateFail');
```

Comment:
@puskin94 I don't understand; what I currently have seems readable... no?

If I do something like this =>

```
async function handleDActions(DBaction, actionType) {
    try {
        await DBaction();
    } catch (e) {
        displayToast(`${actionType}Fail`);
        console.log(e);
    }
}
```
I would have to use it like this =>  await handleDActions(action: await Settings.insert(data)}, 'update');

This seems fine with a small action but this will get out of hand. What exactly do you mean?

Reply:
@hcuffy something like the code you wrote, but even simpler: I guess you can avoid passing the `actionType`, if you are able to use `typeOf` or `instanceOf`